src/voice/connection.rs

```diff
@@ -143,40 +165,32 @@ impl Connection {
             .next()
             .ok_or(Error::Voice(VoiceError::HostnameResolve))?;
 
-        // Important to note here: the length of the packet can be of either 4
-        // or 70 bytes. If it is 4 bytes, then we need to send a 70-byte packet
-        // to determine the IP.
-        //
-        // Past the initial 4 bytes, the packet _must_ be completely empty data.
-        //
-        // The returned packet will be a null-terminated string of the IP, and
-        // the port encoded in LE in the last two bytes of the packet.
         let udp = UdpSocket::bind("0.0.0.0:0")?;
 
+        // Follow Discord's IP Discovery procedures, in case NAT tunnelling is needed.
+        let mut bytes = [0; IpDiscoveryPacket::const_packet_size()];
         {
-            let mut bytes = [0; 70];
-
-            (&mut bytes[..]).write_u32::<BigEndian>(ready.ssrc)?;
-            udp.send_to(&bytes, destination)?;
+            let mut view = MutableIpDiscoveryPacket::new(&mut bytes[..])
+                .expect("[Voice] There are enough bytes to encode a discovery packet.");
```

Comment:
This will panic stating no issue. Is that intended?

Reply:
Yes: the `bytes` is sized to have exactly enough space for an IpDiscovery packet, so failure *cannot* happen. I'm using `expect` when we can make assurances ahead of time (the `discortp`   library can only fail if there are too few bytes).

Reply:
The use of expect is fine, but the message should say something like too few bytes were passed then as it will only show that string of that is the case. If the code is ever changed to cause the expect to trigger.